exporter/solarwindsexporter/config.go

```diff
@@ -68,12 +42,45 @@ type Config struct {
 	QueueSettings exporterhelper.QueueConfig `mapstructure:"sending_queue"`
 	// Timeout configures timeout in the underlying OTLP exporter.
 	Timeout exporterhelper.TimeoutConfig `mapstructure:"timeout,squash"`
+	// ingestionToken stores the token provided by the Solarwinds Extension.
+	ingestionToken configopaque.String `mapstructure:"-"`
+	// endpointURL stores the URL provided by the Solarwinds Extension.
+	endpointURL string `mapstructure:"-"`
 }
 
-// NewDefaultConfig creates a new default configuration.
+// ExtensionAsComponent tries to parse `extension` value of the form 'type/name'
+// or 'type' from the configuration to [component.ID].
+// It fails with an error if it doesn't follow this form or the 'type' part
+// is not a valid [component.Type].
 //
-// Warning: it doesn't define mandatory `Token` and `DataCenter`
-// fields that need to be explicitly provided.
+// Safety: it PANICS if `extension` is empty.
+func (cfg *Config) ExtensionAsComponent() (component.ID, error) {
```

Comment:
do we need this function public?

Reply:
@david-sevcik Great point, yes, it's better to use that one (it seems, they handle same additional) edge cases. I knew they must have it implemented but missed this one can be used by us.

@adschr I think it can be public but let's limit the API surface.

Reply:
Changed in https://github.com/solarwinds/solarwinds-otel-collector/pull/4/commits/4b11ae00436867cf45c6ca5f9bd257fb89fdbb15 (the behaviour seems to be identical and it simplified empty value handling too).